physics/SFC_Models/Land/Noahmp/lnd_iau_mod.F90

```diff
@@ -0,0 +1,915 @@
+   Land_IAU_Control%do_land_iau = do_land_iau
+   Land_IAU_Control%iau_delthrs = land_iau_delthrs
+   Land_IAU_Control%iau_inc_files = land_iau_inc_files
+   Land_IAU_Control%iaufhrs = land_iau_fhrs   
+   Land_IAU_Control%iau_filter_increments = land_iau_filter_increments
+   Land_IAU_Control%lsoil_incr = lsoil_incr
+
+   Land_IAU_Control%me = me
+   Land_IAU_Control%mpi_root = mpi_root
+   Land_IAU_Control%isc = isc
+   Land_IAU_Control%jsc = jsc
+   Land_IAU_Control%nx = nx
+   Land_IAU_Control%ny = ny
+   Land_IAU_Control%tile_num = tile_num
+   Land_IAU_Control%nblks = nblks
+   Land_IAU_Control%lsoil = lsoil
+   Land_IAU_Control%lsnow_lsm = lsnow_lsm
+   Land_IAU_Control%dtp = dtp
+   Land_IAU_Control%fhour = fhour
+
+   Land_IAU_Control%upd_stc = land_iau_upd_stc
+   Land_IAU_Control%upd_slc = land_iau_upd_slc
+   Land_IAU_Control%do_stcsmc_adjustment = land_iau_do_stcsmc_adjustment
+   Land_IAU_Control%min_T_increment = land_iau_min_T_increment
+
+   allocate(Land_IAU_Control%blksz(nblks))
+   allocate(Land_IAU_Control%blk_strt_indx(nblks))
+
+   ! Land_IAU_Control%blk_strt_indx: start index of each block, for flattened (ncol=nx*ny) arrays 
+   ! required in noahmpdriv_run to get subsection of the stc array for each
+   ! proces/thread
+   ix = 1
+   do nb=1, nblks
+      Land_IAU_Control%blksz(nb) = blksz(nb)
+      Land_IAU_Control%blk_strt_indx(nb) = ix
+      ix = ix + blksz(nb)
+   enddo
+
+end subroutine land_iau_mod_set_control
+
+subroutine land_iau_mod_init (Land_IAU_Control, Land_IAU_Data, Land_IAU_State, errmsg, errflg)    
+   type (land_iau_control_type),       intent(inout) :: Land_IAU_Control
+   type (land_iau_external_data_type), intent(inout) :: Land_IAU_Data
+   type(land_iau_state_type),          intent(inout) :: Land_IAU_state  
+   character(len=*),                   intent(  out) :: errmsg
+   integer,                            intent(  out) :: errflg
+
+   ! local
+   character(len=128)   :: fname
+   real(kind=kind_phys) :: sx, wx, wt, normfact, dtp
+   integer              :: k, nstep, kstep
+   integer              :: nfilesall, ntimesall
+   integer, allocatable :: idt(:) 
+   integer              :: nlon, nlat
+   logical              :: exists
+   integer              :: ncid, dimid, varid, status, IDIM
+   
+   real(kind=kind_phys) :: dt   !, rdt
+   integer              :: im, jm, km, nfiles, ntimes
+
+   integer :: is,  ie,  js,  je
+   integer :: npz
+   integer :: i, j  
+
+   !Errors messages handled through CCPP error handling variables
+   errmsg = ''
+   errflg = 0
+
+   npz = Land_IAU_Control%lsoil
+   km = Land_IAU_Control%lsoil
+   
+   is  = Land_IAU_Control%isc
+   ie  = is + Land_IAU_Control%nx-1
+   js  = Land_IAU_Control%jsc
+   je  = js + Land_IAU_Control%ny-1
+   nlon = Land_IAU_Control%nx
+   nlat = Land_IAU_Control%ny
+   !nblks = Land_IAU_Control%nblks
+   !blksz = Land_IAU_Control%blksz(1)
+
+   print*, "rank is ie js je nlon nlat", Land_IAU_Control%me, is, ie, js, je, nlon, nlat

```

Comment:
is this a debug statement? if not, consider making it a bit more verbose to describe what routine this is coming from?